lib/crypto/base10.js

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const assert = require('assert')
+const random = require('./random')
+
+const ALPHABET = '0123456789'
+
+// some sanity checks, hard to test, private to this mdoule
+assert.equal(ALPHABET.length, 10, 'ALPHABET is 10 digits')
+
+function base10(len) {
+  return random(len)
+    .then(bytes => {
+      const out = []
+
+      for (let i = 0; i < len; i++) {
+        out.push(ALPHABET[bytes[i] % 10])
```

Comment:
It seems like the implementation of `base10` is fundamentally the same as the implementation of `base32` in the same directory. You could parameterise the alphabet as an arg to a more generic function and then do this modulo operation by `alphabet.length`.

Less code, more generic, one set of common tests, seems like a pretty easy win?